eogrow/utils/fs.py

```diff
@@ -61,19 +65,19 @@ def __init__(
             self._local_path = fs.path.basename(self._remote_path)
 
         self._absolute_local_path = self._filesystem.getsyspath(self._local_path)
+        self._remote_location_ensured = False
```

Comment:
what is the usecase where this results in an improvement?

Reply:
Something like this:

```Python
with LocalFile("/path/to/file/something.json", mode="w") as local_file:
    # At this point we don't want that path to file is generated yet
    ...
    local_file.copy_to_remote()  # Now the path should be generated
    ...
    local_file.copy_to_remote()  # Path already exists so we don't even have to make IO calls to ensure it exists

...
# Once local_file is closed the file is copied again but again we don't have to ensure that the path exists
```

So basically it aims to reduce a few unnecessary IO calls to remote filesystem.

Reply:
but do we want the user to manually copy the file over multiple times?

Reply:
In some cases, yes. In our case, we use this for copying logs to a remote filesystem every minute or so. This way, logs are available for users to see right away and if at any time an EC2 instance dies the logs are already on S3.

Reply:
using it in logging is reason enough, good point